### PR TITLE
Alphabetically sorts hairstyles & appearance things

### PR DESCRIPTION
**Pull request by Ty-Omaha**
## What Does This PR Do
Sorts all the hairstyles and accessories in alphabetical order.

this is a direct result of me getting angry because I couldn't find if there were dreadlocks or not.

goes with #13038

🆑 
tweak: Alphabetically sorts Appearance list items such as hairstyles
/ 🆑

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Any chance we can get the scissors menu alphabetised? IIRC it isnt at the moment/

**Comment by Ty-Omaha** [comment]
I think would need a whole refactor since from a quick glance it appears to just hardcode the code of the procs that are changed in this PR instead of just using them, will look more into it later

https://github.com/ParadiseSS13/Paradise/blob/3fee0f2019bb83f27aca6f44c0f0770187a7ed59/code/game/objects/items/weapons/scissors.dm

**Comment by Mitchs98** [comment]
I love you.

File: code/modules/client/preference/preferences.dm [hunk 1]
```diff
@@ -1308,7 +1308,7 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 					else //Not using the whitelist? Aliens for everyone!
 						new_species += GLOB.whitelisted_species
 
-					species = input("Please select a species", "Character Generation", null) in new_species
+					species = input("Please select a species", "Character Generation", null) in sortTim(new_species, /proc/cmp_text_asc)
```
Comment (farie82) [inline code/modules/client/preference/preferences.dm]: I think you can just use sortList() instead of these. Same effect with less overhead

**Comment by taukausanake** [comment]
Can we at least get the bald/shaven options first in the lists? It's annoying having to look through the list, even if in order, to find the blank state

**Comment by Ty-Omaha** [comment]
@taukausanake the way it works is sorting by alphabetical order, when it encounters numbers it puts them first. doing that would require snowflake code

File: code/modules/client/preference/preferences.dm (hunk 1 above)
Comment (Ty-Omaha) [reply code/modules/client/preference/preferences.dm]: I can, but sortTim(uses Timsort) is more optimized compared to sortList(Mergesort)

**Comment by Ty-Omaha** [comment]
adjusting scissors is out of scope for this PR they can get their own refactor PR

**Comment by Ty-Omaha** [comment]
@taukausanake why are you downvoting that? you want me to add a specific thing that says "if list item == "bald" then put at front" that will run everytime anything in the code uses sortTim? 😐

**Comment by taukausanake** [comment]
Downvote means I disagree. It's annoying having to try to find bald or shaven in these lists. You'd expect the blank states to be at the top. That's all I'm saying.

_People actually pay attention to the votes?_ I didn't know we were a democracy now /s

**Comment by Crazylemon64** [comment]
@Ty-Omaha the way you'd implement that would be to exclude "bald" from the list beforehand, and then prepend it to the list at the point of return

**Comment by Ty-Omaha** [comment]
but you'd have to use an if statement to exclude it from the list

**Comment by Ty-Omaha** [comment]
unless theres some other way

**Pull request merged**